File: python/retrieving.py
#!/usr/bin/env python
from __future__ import print_function

from couchbase.bucket import Bucket
from couchbase.exceptions import NotFoundError

cb = Bucket('couchbase://10.0.0.31/default')

print('Getting non-existent key. Should fail..')
try:
    cb.get('non-exist-document')
except NotFoundError:
    print('Got exception for missing document!')
print('...')

print('Upserting...')
cb.upsert('new_document', {'foo': 'bar'})
print('Getting...')
print(cb.get('new_document').value)

File: php/retrieving.php
<?php
$cluster = new \Couchbase\Cluster("couchbase://localhost");
$bucket = $cluster->openBucket("default");

echo "Getting non-existent key. Should fail\n";
try {
    $bucket->get('non-exist-document');
} catch (\Couchbase\Exception $ex) {
    if ($ex->getCode() != COUCHBASE_KEY_ENOENT) {
        throw new Exception("GRRR");
    }
    printf("Error: %s (0x%x)\n", $ex->getMessage(), $ex->getCode());
}

echo "Upserting...\n";
$bucket->upsert("new_document", array("foo" => "bar"));
echo "Getting\n";
$result = $bucket->get("new_document");
var_dump($result);
echo "Foo is: " . $result->value->foo . "\n";
